src/translator/service.cpp

```diff
@@ -54,7 +55,7 @@ void Service::build_translators(Ptr<Options> options, size_t numTranslators) {
   translators_.reserve(numTranslators);
   for (size_t cpuId = 0; cpuId < numTranslators; cpuId++) {
     marian::DeviceId deviceId(cpuId, DeviceType::cpu);
-    translators_.emplace_back(deviceId, vocabs_, options, model_memory_);
+    translators_.emplace_back(deviceId, vocabs_, options, std::move(modelMemory_), std::move(shortlistMemory_));
```

Comment:
Service owns the memory. BatchTranslator gets reference collected as pointer.

Only the first BatchTranslator inits here. Others encounter illegal memories. 

I think. Run and check? More than 0 threads variations of service cli.

Reply:
Agree. It was my misunderstanding here.